crates/cargo-contract/src/cmd/extrinsics/runtime_api/mod.rs

```diff
@@ -17,4 +17,22 @@
 #[subxt::subxt(
     runtime_metadata_path = "src/cmd/extrinsics/runtime_api/contracts_runtime.scale"
 )]
-pub mod api {}
+pub mod api {
+    #[subxt(substitute_type = "frame_support::weights::weight_v2::Weight")]
+    use crate::cmd::extrinsics::runtime_api::Weight;
+}
+
+/// Copy of the `weight_v2::Weight` type defined in substrate.
+///
+/// Allows for local trait impls.
+#[derive(scale::Encode, scale::Decode, scale::CompactAs, Debug)]
+pub struct Weight {
+    /// The weight of computational time used based on some reference hardware.
+    ref_time: u64,
+}
+
+impl From<u64> for Weight {
+    fn from(ref_time: u64) -> Self {
+        Self { ref_time }
+    }
+}
```

Comment:
I don't think we should do that given that we expect further fields will be added. I think adding a `from_ref_time` would be more appropriate.